src/get-started/java-binding.md

```diff
@@ -326,6 +377,27 @@ for the given transaction. An observable result is the one that affects the
 blockchain state hash: a modification of a collection that affects the service
 state hash, or an execution exception.
 
+<!-- todo: Shall we put under "Transactions" (as a modifying operation)
+or under "Blockchain Events" (as a handler of 'before commit' event)? -->
+
+#### Block Commit Handler
+
+Services can receive a notification after all transactions in a block has been
+processed but before the block has been committed, allowing inspecting
+the changes made by the transactions and modifying any service state.
+
+Exonum delivers this notification to implementations
+of [`Service#beforeCommit`][service-before-commit] method. The method will see
+the state of the database after all transactions have been applied.
+It may make any changes to the service state, which will be included
+in the block. As transactions, the implementations must produce
+the same observable result on all nodes of the system. 
+If exceptions occur in this handler, Exonum will roll back any changes made
+to the persistent storage in it. 
+<!-- todo: Shall we specify that: -->
```

Comment:
This is going to change soon: exonum/exonum#1576. The question where the result is saved is quite logical, so it could make sense to specify that it's discarded for now.